cf-key/cf-key-functions.c

```diff
@@ -331,6 +292,60 @@ bool KeepKeyPromises(const char *public_key_file, const char *private_key_file,
     return true;
 }
 
+bool KeepKeyPromises(const char *public_key_file, const char *private_key_file, const int key_size)
+{
+    struct stat statbuf;
+
+    if (stat(public_key_file, &statbuf) != -1)
+    {
+        Log(LOG_LEVEL_ERR, "A key file already exists at %s", public_key_file);
+        return false;
+    }
+
+    if (stat(private_key_file, &statbuf) != -1)
+    {
+        Log(LOG_LEVEL_ERR, "A key file already exists at %s", private_key_file);
+        return false;
+    }
+
+    Log(LOG_LEVEL_INFO, "Making a key pair for CFEngine, please wait, this could take a minute...");
+
+#ifdef OPENSSL_NO_DEPRECATED
+    RSA *pair = RSA_new();
+    BIGNUM *rsa_bignum = BN_new();
+    if (pair != NULL && rsa_bignum != NULL)
+    {
+        BN_set_word(rsa_bignum, RSA_F4);
+        int res = RSA_generate_key_ex(pair, key_size, rsa_bignum, NULL);
+        if (res == 0)
+        {
+            RSA_free(pair);
+            pair = NULL;
+        }
+    }
+    else
+    {
+        RSA_free(pair);
```

Comment:
I take it this is safe to call with `NULL`.?

Reply:
Yes.